operator/pkg/translate/translate.go

```diff
@@ -109,19 +104,133 @@ type Translation struct {
 }
 
 // NewTranslator creates a new translator for minorVersion and returns a ptr to it.
-func NewTranslator(minorVersion version.MinorVersion) (*Translator, error) {
-	f := filepath.Join(TranslateConfigFolder, TranslateConfigPrefix+minorVersion.String()+".yaml")
-	b, err := vfs.ReadFile(f)
-	if err != nil {
-		return nil, fmt.Errorf("could not read translateConfig file %s: %s", f, err)
-	}
-	t := &Translator{}
-	err = yaml.Unmarshal(b, t)
-	if err != nil {
-		return nil, fmt.Errorf("could not Unmarshal translateConfig file %s: %s", f, err)
-	}
-	t.Version = minorVersion
-	return t, nil
+func NewTranslator() *Translator {
+	t := &Translator{
+		Version: oversion.OperatorBinaryVersion.MinorVersion,
+		APIMapping: map[string]*Translation{
+			"Hub":         {OutPath: "global.hub"},
+			"Tag":         {OutPath: "global.tag"},
+			"K8SDefaults": {OutPath: "global.resources"},
+			"Revision":    {OutPath: "revision"},
+			"MeshConfig":  {OutPath: "meshConfig"},
+		},
+		GlobalNamespaces: map[name.ComponentName]string{
+			name.PilotComponentName:     "istioNamespace",
+			name.TelemetryComponentName: "telemetryNamespace",
+			name.PolicyComponentName:    "policyNamespace",
+		},
+		ComponentMaps: map[name.ComponentName]*ComponentMaps{
+			name.IstioBaseComponentName: {
+				HelmSubdir:           "base",
+				ToHelmValuesTreeRoot: "global",
+				SkipReverseTranslate: true,
+			},
+			name.PilotComponentName: {
+				ResourceType:         "Deployment",
+				ResourceName:         "istiod",
```

Comment:
nit: do we have constants available for any of these names?

Reply:
Did a quick grep, looks like we do not. Probably a good followup idea though.